File: 3236-smallest-missing-integer-greater-than-sequential-prefix-sum/smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, nums: list[int]) -> int:
        seqSum = nums[0]
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1] + 1:
                seqSum += nums[i]
            else:
                break

        while True:
            found = False
            for i in range(len(nums)):
                if seqSum == nums[i]:
                    found = True
                    break
            if not found:
                return seqSum
            seqSum += 1

File: 3236-smallest-missing-integer-greater-than-sequential-prefix-sum/smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int seqSum = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == nums[i - 1] + 1) {
                seqSum += nums[i];
            } else {
                break;
            }
        }

        while (true) {
            bool found = false;
            for (int i = 0; i < nums.size(); i++) {
                if (seqSum == nums[i]) {
                    found = true;
                    break;
                }
            }
            if (!found) {
                return seqSum;
            }
            seqSum++;
        }
    }
};